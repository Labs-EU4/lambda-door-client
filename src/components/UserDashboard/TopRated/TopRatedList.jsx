/* eslint-disable no-nested-ternary */
/* eslint-disable no-shadow */
import React, { useEffect } from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { Spin } from 'antd';
import CompanySummaryCard from '../../CompanySummaryCard';
import { getTopRatedReviews } from '../../../state/actions/topRatedReviews';

export const TopRatedList = ({
  isFetching,
  getTopRatedReviews,
  topRatedReviews: { topRatedReviews },
  authState: {
    credentials: { location },
  },
}) => {
  useEffect(() => {
    getTopRatedReviews();
  }, []);
  return (
    <StyledDiv>
      {!isFetching ? (
        <>
          {topRatedReviews.length === 0 ? (
            <div className="empty-state">
              <p>No data to display</p>
            </div>
          ) : !location ? (
            topRatedReviews.slice(0, 8).map(topRated => (
              // eslint-disable-next-line react/no-array-index-key
              <CompanySummaryCard
                key={`${topRated.id}`}
                text={topRated.description}
                name={topRated.name}
                id={topRated.id}
                rating={topRated.average_rating}
                website={topRated.website}
              />
            ))
          ) : (
            topRatedReviews.slice(0, 4).map(topRated => (
              // eslint-disable-next-line react/no-array-index-key
              <CompanySummaryCard
                key={`${topRated.id}`}
                text={topRated.description}
                name={topRated.name}
                id={topRated.id}
                rating={topRated.average_rating}
                website={topRated.website}
              />
            ))
          )}
        </>
      ) : (
        <div className="empty-state">
          <Spin />
        </div>
      )}
    </StyledDiv>
  );
};

export default connect(state => state, { getTopRatedReviews })(TopRatedList);

const StyledDiv = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 1.5rem;
  position: relative;
  min-height: 300px;

  @media (max-width: 768px) {
    grid-template-columns: repeat(1, 1fr);
  }

  .cards {
    position: relative;
    height: 150px;
    max-width: 350px;

    a {
      position: absolute;
      left: 0;
      right: 0;
      top: 0;
      bottom: 0;
      padding: 1rem;
      color: rgba(0, 0, 0, 0.65);
      border: 1px solid #e8e8e8;
      border-radius: 2px;
      display: flex;
      flex-direction: column;
      justify-content: space-between;

      &:hover {
        border: 1px solid #bb1333;
      }
    }

    p {
      height: 40px;
      overflow: hidden;
    }

    h3 {
      font-weight: 600;
      font-size: 1rem;
    }

    .ant-rate,
    .ant-rate-star-first,
    .ant-rate-star-second {
      height: 20px;
      /* width: fit-content;
      display: inline-flex;
      align-items: center;
      justify-content: center; */
    }

    .ant-rate-star-first i,
    .ant-rate-star-second i {
      font-size: 14px;
    }

    .ant-rate-star:not(:last-child) {
      margin-right: 3px;
    }
  }
`;
