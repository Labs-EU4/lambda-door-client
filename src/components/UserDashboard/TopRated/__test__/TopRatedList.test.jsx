import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { TopRatedList } from '../TopRatedList';

const getTopRatedReviews = jest.fn();
const isFetching = false;
const topRatedReviews = {
  topRatedReviews: [
    {
      user_id: 1,
      company_id: 10,
      ratings: 4,
      is_currently_employed: 1,
      review_headline: 'Great Company to work for',
      review:
        'Extremely good benefits from healthcare, PTO, and discounts. Very relaxed environment with clearly laid out expectations.',
      is_accepting_questions: 0,
    },
  ],
};
let wrapper;
beforeEach(() => {
  rtl.cleanup();
  wrapper = rtl.render(
    <Router>
      <TopRatedList
        getTopRatedReviews={getTopRatedReviews}
        isFetching={isFetching}
        topRatedReviews={topRatedReviews}
      />
    </Router>
  );
});

describe('TopRatedList', () => {
  wrapper = rtl.render(
    <Router>
      <TopRatedList
        getTopRatedReviews={getTopRatedReviews}
        isFetching={isFetching}
        topRatedReviews={topRatedReviews}
      />
    </Router>
  );
  it('renders correctly', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});


