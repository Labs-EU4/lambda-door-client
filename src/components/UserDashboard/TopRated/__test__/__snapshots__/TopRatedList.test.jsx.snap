// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopRatedList renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bxivhb mdOKc"
    >
      <div
        class="cards"
      >
        <a
          class="nav-link"
          href="/company-page/undefined"
        >
          <div>
            <h3 />
            <p
              class="sc-bwzfXH hKkmqJ"
            />
          </div>
          <div
            class="sc-htpNat hiPvMK"
          >
            <div />
            <div />
          </div>
        </a>
      </div>
    </div>
  </div>
</body>
`;
